controller/scheduler-v2/scheduler.go

```diff
@@ -209,9 +212,19 @@ func (s *Scheduler) SyncFormations() (err error) {
 }
 
 func (s *Scheduler) RectifyJobs() (err error) {
-	log := s.log.New("fn", "RectifyJobs")
+	s.leaderMtx.RLock()
+	defer s.leaderMtx.RUnlock()
```

Comment:
We should handle the leader demotion using a channel in the main loop and avoid this mutex.